test/test_integration.py

```diff
@@ -324,3 +327,140 @@ def test_gallery(self):
             '#dropdown .Select-input input'
         ).send_keys(u'北')
         self.snapshot('gallery - chinese character')
+
+    def test_location_link(self):
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            html.Div(id='waitfor'),
+            dcc.Location(id='test-location', refresh=False),
+
+            dcc.Link(
+                html.Button('I am a clickable button'),
+                id='test-link',
+                href='/test/pathname'),
+            dcc.Link(
+                html.Button('I am a clickable hash button'),
+                id='test-link-hash',
+                href='#test'),
+            dcc.Link(
+                html.Button('I am a clickable search button'),
+                id='test-link-search',
+                href='?testQuery=testValue',
+                refresh=False),
+            html.Button('I am a magic button that updates pathname', id='test-button'),
+            html.A('link to click', href='/test/pathname/a', id='test-a'),
+            html.A('link to click', href='#test-hash', id='test-a-hash'),
+            html.A('link to click', href='?queryA=valueA', id='test-a-query'),
+            html.Div(id='test-pathname', children=[]),
+            html.Div(id='test-hash', children=[]),
+            html.Div(id='test-search', children=[]),
+        ])
+
+        @app.callback(
+            output=Output(component_id='test-pathname', component_property='children'),
+            inputs=[Input(component_id='test-location', component_property='pathname')])
+        def update_location_on_page(pathname):
+            assert isinstance(pathname, str)
```

Comment:
I believe that these asserts won't actually fail the test as the dash app is run in a subprocess in these tests. They'll still cause the callback to fail (which might implicitly break other assertions in the main process) but in either case we might want to put this assertion out of this context